internal/querynodev2/segments/result.go

```diff
@@ -285,6 +286,7 @@ func MergeInternalRetrieveResult(ctx context.Context, retrieveResults []*interna
 			typeutil.AppendPKs(ret.Ids, pk)
```

Comment:
here, we only accumulate j when there is really one more result appended, if it's a duplicated pk, then not accumulate j